Add versioning section and refactor the readme (#378)
oam-dev/spec: README.md

## Introduction

> The motivation behind Open Application Model is to provide a standard but flexible framework for platform builders to create application centric platforms based on lower level orchestration systems like Kubernetes, in approach with discoverability, manageability and interoperability. 

Reviewer: ```suggestion
Open Application Model provides a standard and flexible framework for platform builders to create application centric platforms on top of infrastructure orchestration systems such as Kubernetes with additions of discoverability, manageability and interoperability. 
```

![How it works][how-it-works]

> Open Application Model enables the platforms expose standardized application centric primitives (e.g. Workloads and Traits) instead of leaking infrastructure details to end user, and at the same time, keeps the extensibility of the underlying orchestration system. Though with the idea of making creating upper layer platforms easier, the model itself won't lock you in arbitrary abstractions, on the contrary, its primitives allow you to define the right level of abstractions by yourself.

Reviewer 2: ```suggestion
Open Application Model enables the platforms expose standardized application centric primitives (e.g. Workloads and Traits) instead of leaking infrastructure details to end users, and at the same time, keeps the extensibility of the underlying orchestration system. Though with the idea of making creating upper layer platforms easier, the model itself won't lock you in arbitrary abstractions, on the contrary, its primitives allow you to define the right level of abstractions by yourself.
```

> Open Application Model enables the platforms expose standardized application centric primitives (e.g. Workloads and Traits) instead of leaking infrastructure details to end users, and at the same time, keeps the extensibility of the underlying orchestration system. Though with the idea of making creating upper layer platforms easier, the model itself won't lock you in arbitrary abstractions, on the contrary, its primitives allow you to define the right level of abstractions by yourself.

Reviewer: ```suggestion
Open Application Model empowers application platforms to provide standardized application centric primitives (e.g. Workloads and Traits) instead of exposing infrastructure details to end users., while retaining the extensibility of the underlying orchestration system. With the idea of simplify creating upper layer platforms, the model won't lock you into specific abstractions -- on the contrary, its primitives allow you to define the right level of abstractions by yourself.
```

### A team-centric model

> When it comes to the application centric primitives, we think it is important to distinguish between the parts that developers are responsible for, and the parts that operations is responsible for. Otherwise, if these roles get muddled, it would result in communications mishaps, bugs, or even service outages.

Reviewer 2: ```suggestion
When it comes to the application centric primitives, we think it is important to distinguish between the parts that developers are responsible for, and the parts that operations are responsible for. Otherwise, if these roles get muddled, it would result in communications mishaps, bugs, or even service outages.
```
And `operations` here means `infrastructure operators` and `application operators`?

Author: It should be a typo

* _Application Operators_ are responsible for configuring the runtime aspects of
  one or more of these microservices. They are the domain experts on the
  platform.
* _Infrastructure Operators_ also known as _Platform Builders_, are responsible for setting up and maintaining the
  infrastructure within which applications run. They are the domain
  experts on the low-level details.

For more details and user stories, see [introduction.md](./introduction.md).

## See it in action

[OAM Kubernetes Runtime](https://github.com/crossplane/oam-kubernetes-runtime) is the officially maintained OAM plugin for Kubernetes. 

> OAM Kubernetes Runtime is a [joint effort](https://cloudblogs.microsoft.com/opensource/2020/05/27/open-application-model-oam-v1alpha2-crossplane/) with [the Crossplane community](https://github.com/crossplane/crossplane). Furthermore, to get started with an example of using OAM to deliver both applications and cloud resources in unified approach, please follow the end-to-end [getting started doc in Crossplane](https://crossplane.io/docs/v0.11/getting-started/run-applications.html).

Reviewer 2: How about we using the latest version `v0.12` for end-to-end documentation?

Author: NP

## The Specification

The specification convention adopts [Kubernetes Resource Model](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/architecture/resource-management.md) which we believe will become the standard interface for the majority of platforms in the future.

  - [Notational Conventions](notational_convention.md)
  - [Purpose and Goals](1.purpose_and_goals.md)
  - [Overview and Terminology](2.overview_and_terminology.md)
  - API Resource Specification
    - Application Developers
      - [Components](4.component.md)
    - Application Operators
      - [Application Scopes](5.application_scopes.md)
      - [Traits](6.traits.md)
      - [Application Configuration](7.application_configuration.md)
    - Infrastructure Operators/Platform Builders
      - [Workload Definition](3.workload.md)
  - [Practical Considerations](8.practical_considerations.md)
  - [Design Principles](9.design_principles.md)

## Community

### Versioning

> Since July 2020, changes to the specification are versioned according to [Semantic Versioning 2.0](https://semver.org/spec/v2.0.0.html) and described in [its release page](https://github.com/oam-dev/spec/releases). Layout changes are not versioned. Specific implementations of the specification (e.g. OAM Kubernetes Runtime ) should specify which version they implement.

Reviewer 2: Could you please help explain a little bit for `Layout changes`?

Author: Doc format, repo organization etc.

> Changes to the change process itself are not currently versioned but may be independently versioned in the future.

Reviewer: What do you mean "Changes to the change process"?

Author: Changes to governance model, to spec review/merge process etc.
